modules/apps/29-fee/module_test.go

```diff
@@ -0,0 +1,245 @@
+package fee_test
+
+import (
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	transfertypes "github.com/cosmos/ibc-go/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/modules/core/24-host"
+)
+
+func (suite *FeeTestSuite) TestOnChanOpenInit() {
+	testCases := []struct {
+		name    string
+		version string
+		expPass bool
+	}{
+		{
+			"valid fee middleware and transfer version",
+			"fee29-1:ics20-1",
+			true,
+		},
+		{
+			"invalid fee middleware version",
+			"otherfee28-1:ics20-1",
+			false,
+		},
+		{
+			"invalid transfer version",
+			"fee29-1:wrongics20-1",
+			false,
+		},
+		{
+			"incorrect wrapping delimiter",
+			"fee29-1//ics20-1",
+			false,
+		},
+		{
+			"transfer version not wrapped",
+			"fee29-1",
+			false,
+		},
+		{
+			"fee version not included",
+			"ics20-1",
+			false,
+		},
+		{
+			"hanging delimiter",
+			"fee29-1:ics20-1:",
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		ctx := suite.chainA.GetContext()
+		cap, err := suite.chainA.GetSimApp().ScopedIBCKeeper.NewCapability(ctx, tc.name)
+		suite.Require().NoError(err)
+		err = suite.module.OnChanOpenInit(
+			ctx,
+			channeltypes.UNORDERED,
```

Comment:
Can we have tests for both unordered and ordered since both should be supported (happy case ensuring that check isn't added later)